### PR TITLE
NATIVE-486: Add documentation for CI/CD

### PR DESCRIPTION
**Pull request by maxammann**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **NATIVE**.

### CODE REVIEW

File: android/fastlane/Fastfile [hunk 1]
```diff
@@ -6,12 +6,12 @@ CREDENTIALS_GIT_REPOSITORY_URL = ENV['CREDENTIALS_GIT_REPOSITORY_URL']
...
-KEYSTORE_KEY_ALIAS = ENV['KEYSTORE_KEY_ALIAS'] || 'test'
-KEYSTORE_PASSWORD = ENV['KEYSTORE_PASSWORD'] || '123456'
-KEYSTORE_KEY_PASSWORD = ENV['KEYSTORE_KEY_PASSWORD'] || '123456'
+KEYSTORE_PATH = ENV['KEYSTORE_PATH']
+KEYSTORE_KEY_ALIAS = ENV['KEYSTORE_KEY_ALIAS']
+KEYSTORE_PASSWORD = ENV['KEYSTORE_PASSWORD']
+KEYSTORE_KEY_PASSWORD = ENV['KEYSTORE_KEY_PASSWORD']
 
-desc "Prepare the keystore"
+desc "Downloads and decrypts the JKS"
```
Comment (klinzo) [inline android/fastlane/Fastfile:14]: I think this is a different naming. Comments normally don't use third person in our project.

File: android/fastlane/README.md [hunk 2]
```diff
@@ -19,17 +19,17 @@ or alternatively using `brew cask install fastlane`
 ```
 fastlane keystore
 ```
-Prepare the keystore
+Downloads and decrypts the JKS
```
Comment (klinzo) [inline android/fastlane/README.md:22]: Same as above, I'd rather use
```suggestion
Download and decrypt the JKS
```

File: docs/04-technical-documentation.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Technical documentation
+## Content
+- [Glossary](16-glossary.md)
```
Comment (klinzo) [inline docs/04-technical-documentation.md:3]:
```suggestion
* [Glossary](16-glossary.md)
```

File: docs/04-technical-documentation.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+* [Error reporting](05-error-reporting.md)
+* [React Native Upgrades](15-react-native-upgrades.md)
+
+## Technology Stack
+### Bundler
+[Metro](https://facebook.github.io/metro/) is used to compile and bundle the app.
+
+### Frontend framework
+[React](https://facebook.github.io/react/) is used as frontend framework.
+This allows us to build a single-page-application easily.
```
Comment (klinzo) [inline docs/04-technical-documentation.md:18]: Probably not accurate for react native.

File: docs/05-error-reporting.md [hunk 5]
```diff
@@ -29,7 +29,7 @@ Only proper releases should be handled correctly by sentry. This means we do no
 
 ## Upload of source maps
 
-Sentry wants source maps in order to display stack traces properly. We are not uploading them automatically. Instead you can [use the cli](https://docs.sentry.io/platforms/javascript/sourcemaps/#uploading-source-maps-to-sentry) to upload source maps.
+Sentry wants source maps in order to display stack traces properly. We are not uploading them automatically. Instead, you can [use the cli](https://docs.sentry.io/platforms/javascript/sourcemaps/#uploading-source-maps-to-sentry) to upload source maps.
```
Comment (klinzo) [inline docs/05-error-reporting.md:32]:
```suggestion
Sentry needs source maps in order to display stack traces properly. We are not uploading them automatically. Instead, you can [use the cli](https://docs.sentry.io/platforms/javascript/sourcemaps/#uploading-source-maps-to-sentry) to upload source maps.
```
Is this even necessary anymore? Isn't it uploaded automatically with each realease?

File: docs/06-manual-builds.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
+# Manual builds
+
+If you want to create `.ipa` or `.apk` builds for testing purposes you can follow this guide. If you only want to quickly push an update to the stores then take a look [at the CI/CD documentation](08-cicd.md#triggering-a-build-in-ci).
```
Comment (klinzo) [inline docs/06-manual-builds.md:3]:
```suggestion
If you want to run the app locally or create `.ipa` or `.apk` builds for testing purposes you can follow this guide. If you only want to quickly push an update to the stores then take a look [at the CI/CD documentation](08-cicd.md#triggering-a-build-in-ci).
```

File: docs/06-manual-builds.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+Setup the certificates locally:
+
+```bash
+cd ios/
+bundle exec fastlane certificates
+```
+
+*Hint: After setting up the certificates you can also star to use XCode to build the app.*
```
Comment (klinzo) [inline docs/06-manual-builds.md:23]: Do we have an example on how to do this?
```suggestion
*Note: After setting up the certificates you can also start to use XCode to build the app.*
```

File: docs/06-manual-builds.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+Build the app:
+
+```bash
+yarn
+cd ios/
+bundle exec pod install
```
Comment (klinzo) [inline docs/06-manual-builds.md:30]: Imo this should be moved to setup since those are duplicates.

File: docs/06-manual-builds.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
+# Manual builds
+
+If you want to create `.ipa` or `.apk` builds for testing purposes you can follow this guide. If you only want to quickly push an update to the stores then take a look [at the CI/CD documentation](08-cicd.md#triggering-a-build-in-ci).
+
+## Manually building for iOS
```
Comment (klinzo) [inline docs/06-manual-builds.md:5]: Imo this section is too complicated if I just want to run it locally on my device/on the emulator (which should be the main usecase here). At least the section to run it in the emulator/local device should be first imo.

Comment (klinzo) [reply docs/06-manual-builds.md:5]: I think the corresponding commands should also be part of getting started of the readme. If I am a new contributor to the project, I don't want to know all the details of fastlane, certificates.... but rather just run the app and start developing.

File: docs/06-manual-builds.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+If you do not have Fastlane installed [skip to the next section](#quick-builds-using-a-test-signing-key).
+Firstly you have to prepare a [few environment variables](08-cicd.md#environment-variables-and-dependencies):
+
+```bash
+export CREDENTIALS_GIT_REPOSITORY_URL=<secret>
+export CREDENTIALS_DIRECTORY_PATH=/tmp/credentials
+export CREDENTIALS_KEYSTORE_PASSWORD=<secret>
+export CREDENTIALS_KEYSTORE_PATH=/tmp/credentials/<secret>.enc
+export KEYSTORE_PATH=/tmp/keystore.jks
```
Comment (klinzo) [inline docs/06-manual-builds.md:67]: I think it would be better not having to define environment variables but rather passing them as arguments to fastlane commands.

File: docs/06-manual-builds.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+Finally, provide some more information about unlocking the JKS:
+
+```bash
+export KEYSTORE_KEY_ALIAS=<secret>
+export KEYSTORE_PASSWORD=<secret>
+export KEYSTORE_KEY_PASSWORD=<secret>
+```
+
+The last step is to build to app:
```
Comment (klinzo) [inline docs/06-manual-builds.md:85]:
```suggestion
The last step is to build the app:
```

File: docs/06-manual-builds.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+Optionally, you can also set the version name and code:
+
+```bash
+export ORG_GRADLE_PROJECT_VERSION_CODE=1
+export ORG_GRADLE_PROJECT_VERSION_NAME=0.1
+```
+
+Then you can create a quick rest build in release mode and run it on your emulator:
```
[the remark this answers is not in the document]
Comment (klinzo) [reply docs/06-manual-builds.md:121]: I'd also move this before any optional setup. As mentioned above, imo this is too complicated.

[the remark this answers is not in the document]
Comment (klinzo) [reply docs/06-manual-builds.md:121]: Also running the device locally on my device is completely missing.

File: docs/08-cicd.md [hunk 13]
```diff
@@ -1,80 +1,190 @@
 # Continuous Integration and Delivery
 
-# Worker limits aka. $TOTAL_CPUS
+# Workflows
 
-# Pipeline Steps
+Several workflows exist for different purposes. See the table below for an overview.
 
-# Branches vs Pull Requests vs Scheduled
+|Workflow|Type-checking|Jest tests|Production build|E2E tests|Delivered in production|
```
Comment (klinzo) [inline docs/08-cicd.md:7]: The term `Production build` suggests there is also such a thing as `Debug build`. Instead, the only difference is setting the environment variable `E2E_TESTING_IDS`.

Comment (klinzo) [reply docs/08-cicd.md:7]:
```suggestion
|Workflow|Type-checking|Jest tests|Production build|E2E tests|Delivered to production|
```

File: docs/08-cicd.md [hunk 14]
```diff
@@ -1,80 +1,190 @@
...
+|api-triggered|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:x:|-|
 
+Whether `api-triggered` is delivered to production depends on the supplied parameters.
 
-# Environment variables and dependencies
+## About the "api-triggered" Workflow 
 
-`DELIVERINO_PRIVATE_KEY` (\n in CircleCI)
+This workflow can be triggered using the tool `yarn trigger-pipeline` as described [below](#triggering-a-build-in-ci).
+Depending on the supplied parameters a weekly or bi-weekly build is triggered.
```
Comment (klinzo) [inline docs/08-cicd.md:19]: This is not correct. There is no such thing as weekly and bi-weekly builds.

File: docs/08-cicd.md [hunk 15]
```diff
@@ -1,80 +1,190 @@
...
 
-`FASTLANE_USER`
+### Triggering a Build in CI
 
-`FASTLANE_PASSWORD`
+If you decide to do a non-scheduled automatic release of the app the easiest way is to trigger a build in CI/CD.
+First you need to get acquire a ["Personal API token"](https://circleci.com/docs/2.0/managing-api-tokens/).
```
Comment (klinzo) [inline docs/08-cicd.md:24]:
```suggestion
First you need to get a ["Personal API token"](https://circleci.com/docs/2.0/managing-api-tokens/).
```

File: docs/08-cicd.md [hunk 16]
```diff
@@ -1,80 +1,190 @@
...
+Then you can trigger a build using the tool `yarn trigger-pipeline`. If you execute it without parameters you will see the usage.
```
Comment (klinzo) [inline docs/08-cicd.md:25]: Is there a documentation for this script? If so, this and the lines above should be moved to the corresponding section.

File: docs/08-cicd.md [hunk 17]
```diff
@@ -1,80 +1,190 @@
...
-`export CREDENTIALS_KEYSTORE_PATH=/tmp/credentials/<secret>.enc`
-`export KEYSTORE_PATH=/tmp/keystore.jks`
-`fastlane keystore`
+## Setup of Fastlane
 
-`export KEYSTORE_KEY_ALIAS=<secret>`
-`export KEYSTORE_PASSWORD=<secret>`
-`export KEYSTORE_KEY_PASSWORD=<secret>`
-`fastlane build`
+* [Install Ruby >= 2.6.5](https://www.ruby-lang.org/en/documentation/installation/) (The preferred and tested way is using [Ruby Version Manager (RVM)](https://rvm.io/))
```
Comment (klinzo) [inline docs/08-cicd.md:34]:
```suggestion
* Install [Ruby >= 2.6.5](https://www.ruby-lang.org/en/documentation/installation/) (The preferred and tested way is to use [Ruby Version Manager (RVM)](https://rvm.io/)).
```

File: docs/08-cicd.md [hunk 18]
```diff
@@ -1,80 +1,190 @@
...
+  * If using RVM you have to do: `rvm use ruby-2.6.5`
+* Make sure `ruby --version` reports the correct version
+* Run `bundle install --path vendor/bundle` in the project root, `./android/` and `./ios/`
```
Comment (klinzo) [inline docs/08-cicd.md:37]: Is it really required to run this command thrice?

File: docs/08-cicd.md [hunk 19]
```diff
@@ -1,80 +1,190 @@
...
 
-`FASTLANE_PASSWORD`
+If you decide to do a non-scheduled automatic release of the app the easiest way is to trigger a build in CI/CD.
+First you need to get acquire a ["Personal API token"](https://circleci.com/docs/2.0/managing-api-tokens/).
+Then you can trigger a build using the tool `yarn trigger-pipeline`. If you execute it without parameters you will see the usage.
 
-# Manual deployment
 
-## iOS
+# Fastlane
```
Comment (klinzo) [inline docs/08-cicd.md:28]: When do I need to execute these commands?

File: docs/08-cicd.md [hunk 20]
```diff
@@ -1,80 +1,190 @@
...
-Access to the Bot is granted by a Private Key in PEM format. This is used to get an access token for an installation. This access_token allows to write content to the repositories/organisations where it was installed.
+### Android
 
-// PEM is base64 encoded
-// Disable "Include Administrators" in Protected Branches (GitHub App is Admin)
+For an overview of the Fastlane lanes refer to the [auto-generated README](../android/fastlane/README.md#available-actions). These lanes are responsible for setting up the signing keys and building the Android app.
 
+### iOS
+
+For an overview of the Fastlane lanes refer to the [auto-generated README](../ios/fastlane/README.md#available-actions). These lanes are responsible for setting up the certificates and building the iOS app.
```
Comment (klinzo) [inline docs/08-cicd.md:56]: Duplicates

File: docs/08-cicd.md [hunk 21]
```diff
@@ -1,80 +1,190 @@
...
+
+# Setup Certificates for iOS on Mac
+
+The setup of certificates is demonstrated in section [Manual builds](06-manual-builds.md#manually-building-for-ios).
```
Comment (klinzo) [inline docs/08-cicd.md:60]: If sections are used in multiple docs, they should probably be moved to a separate file. I don't see an order why some stuff is explained in manual builds and some stuff in cicd.

File: docs/08-cicd.md [hunk 22]
```diff
@@ -1,80 +1,190 @@
...
+|Development|:heavy_check_mark:|:x:|:x:|
+|Ad-hoc|:x:|:heavy_check_mark:|:x:|
+|Enterprise|:x:|:heavy_check_mark:|:heavy_check_mark:|
+|App Store|:x:|:x:|:heavy_check_mark:|
+
+Also, there are various ways to install an iOS app:
+
+* Via a URL which is opened in Safari
+* Via App Store
+* Via XCode
```
Comment (klinzo) [inline docs/08-cicd.md:77]: Does not really relate to available certificates

File: docs/08-cicd.md [hunk 23]
```diff
@@ -1,80 +1,190 @@
...
+
+Development certificates are for developers who want to install the app on their devices. You have to select the devices in the Provisioning Profile at [developer.apple.com](https://developer.apple.com/account/resources/profiles/list) you want to install the app on. This means the Device ID (UDID) has to be whitelisted.
```
Comment (klinzo) [inline docs/08-cicd.md:79]:
```suggestion
Development certificates are for developers who want to install the app on their devices. You have to select the devices in the [Provisioning Profile](https://developer.apple.com/account/resources/profiles/list) you want to install the app on. This means the Device ID (UDID) has to be whitelisted.
```

File: docs/08-cicd.md [hunk 24]
```diff
@@ -1,80 +1,190 @@
...
+
+The setup of the JKS is demonstrated in section [Manual builds](06-manual-builds.md#manually-building-for-android).
+
+# Determine the Next Version
+
+The next version of the app must be determined programmatically. The tool `yarn bump-version` can be used. If you execute it without parameters you will see the usage.
+
+As the version number is based on the release date, the first versions of 2020 are `2020.1.0`, `2020.1.1`, `2020.1.2`...
+
+The first release february will have the version number `2020.2.0`.
```
Comment (klinzo) [inline docs/08-cicd.md:97]:
```suggestion
The following schema is used for versioning: `YYYY.mm.<release_number>`, where `<release_number>` is a counter for the number of releases in the corresponding month. E.g. the first release in January 2020 uses `2020.01.1`, the second `2020.01.2`.
```

File: docs/08-cicd.md [hunk 25]
```diff
@@ -1,80 +1,190 @@
...
+As the version number is based on the release date, the first versions of 2020 are `2020.1.0`, `2020.1.1`, `2020.1.2`...
+
+The first release february will have the version number `2020.2.0`. 
+ 
+
+# Environment Variables and Dependencies
+
+|Variable|Description|Where do I get it from?|Example|Reference|
+|---|---|---|---|---|
+|BROWSERSTACK_ACCESS_KEY|Access Key for BrowserStack|Password Manager|steffen|[Appium REST API](https://www.browserstack.com/app-automate/rest-api)|
```
Comment (klinzo) [inline docs/08-cicd.md:104]: I don't think `steffen` is an access key :thinking:

File: docs/14-i18n.md [hunk 26]
```diff
@@ -27,7 +27,7 @@ UI elements are:
 The content language is changeable in the app and is set to the UI language at the first app start.
 
 All content elements are using the content language and the corresponding layout direction, i.e. LTR or RTL.
-Content elements are all elements that are NOT UI elements, especially
+Content elements are all elements that are NOT UI elements, especially.
```
Comment (klinzo) [inline docs/14-i18n.md:30]:
```suggestion
Content elements are all elements that are NOT UI elements, especially:
```

File: docs/16-windows-setup.md [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+If react native is already running on your machine you can skip this part.
+
+This guide will help you setup a fresh development environment if you have no development tools on your machine.
+I recommend [Chocolatey](https://chocolatey.org/) as package manager for the tool installation, but is not required. 
+This is usually a one-click solution and very easy to use. If you don't use choco you need to install the listed tools manually.
+
+To install Chocolatey open PowerShell as **admin** and run:
+ 
+ ````shell script
+  Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
```
Comment (klinzo) [inline docs/16-windows-setup.md:16]: :thinking:

File: fastlane/Fastfile [hunk 28]
```diff
@@ -58,7 +58,7 @@ end
 # new_version_name: The new version name
 # package_name: The name of the app in Sentry
 # sourcemap_path: The path to the sourcemap file to upload
-desc "Upload source map to Sentry"
+desc "Uploads a source map to Sentry"
```
Comment (klinzo) [inline fastlane/Fastfile:61]: Why this new naming? Haven't used that anywhere yet.

File: docs/08-cicd.md (hunk 24 above)
Comment (maxammann) [reply docs/08-cicd.md]: We start at 0. I modified the docu here.

File: docs/06-manual-builds.md (hunk 10 above)
Comment (maxammann) [reply docs/06-manual-builds.md:92]: I created https://issues.integreat-app.de/browse/NATIVE-548 instead

File: docs/08-cicd.md (hunk 21 above)
Comment (maxammann) [reply docs/08-cicd.md]: I moved this to a separate file

File: docs/06-manual-builds.md (hunk 9 above)
Comment (maxammann) [reply docs/06-manual-builds.md:5]: I tried to restructure it a bit. But it still is a little bit complicated.

I already linked it from the README.

File: README.md [hunk 29]
```diff
@@ -1,108 +1,76 @@
...
-Note: All dependencies are handled by CocoaPods. The versions in node_modules/ should not be used!
+### Project setup
+We suggest **[IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/)** and XCode as IDE. JetBrains provides free licences for students.
+If you are using a different IDE, the steps might differ.
 
-Note: If you are using Xcode, always open `project.xcworkspace`
+* Import this project (VCS > Get from Version Control).
+* Run `yarn` in the terminal to install all dependencies.
+* Take a look at package.json to show all available npm scripts.
+* Run `yarn start` to start the bundler.
```
[the remark this answers is not in the document]
Comment (maxammann) [reply README.md:45]: This is just for the JS part. The platform specific doucmentation is in Manual Builds

File: docs/06-manual-builds.md [hunk 30]
```diff
@@ -0,0 +1,161 @@
...
+```
+
+Setup your JKS now:
+
+```bash
+cd android/
+bundle exec fastlane keystore
+```
+
+Finally, provide some more information about unlocking the JKS:
```
Comment (klinzo) [inline docs/06-manual-builds.md:102]: I'd add these to prepare environment variables above.

File: docs/06-manual-builds.md [hunk 31]
```diff
@@ -0,0 +1,161 @@
...
+cd android/
+bundle exec fastlane build
+```
+
+Install and run the app using:
+
+```bash
+adb install app/build/outputs/apk/release/app-release.apk
+adb shell am force-stop tuerantuer.app.integreat
+adb shell am start -n tuerantuer.app.integreat/.MainActivity
```
Comment (klinzo) [inline docs/06-manual-builds.md:123]: Why is this necessary?

File: docs/06-manual-builds.md [hunk 32]
```diff
@@ -0,0 +1,161 @@
...
+cd android/
+bundle exec fastlane keystore
+```
+
+Finally, provide some more information about unlocking the JKS:
+
+```bash
+export KEYSTORE_KEY_ALIAS=<secret>
+export KEYSTORE_PASSWORD=<secret>
+export KEYSTORE_KEY_PASSWORD=<secret>
```
Comment (klinzo) [inline docs/06-manual-builds.md:107]: I'd move this to a separate sub like in the ios part. I'd also apply the suggestions made for the ios certificate setup part.

File: docs/06-manual-builds.md [hunk 33]
```diff
@@ -0,0 +1,161 @@
...
+Then you can create a quick test build in release mode and run it on your emulator:
+
+```bash
+yarn
+yarn android:release
+```
+
+### Run the app in an Emulator
+
+If you want to quickly run the app in an emulator do the following:
```
Comment (klinzo) [inline docs/06-manual-builds.md:155]:
```suggestion
### Run the app in an Emulator or device

If you want to quickly run the app in an emulator or connected device do the following:
```

File: docs/08-cicd.md [hunk 34]
```diff
@@ -1,80 +1,154 @@
...
+## Setup of Fastlane
 
-`export KEYSTORE_KEY_ALIAS=<secret>`
-`export KEYSTORE_PASSWORD=<secret>`
-`export KEYSTORE_KEY_PASSWORD=<secret>`
-`fastlane build`
+* Install [Ruby >= 2.6.5](https://www.ruby-lang.org/en/documentation/installation/) (The preferred and tested way is to use [Ruby Version Manager (RVM)](https://rvm.io/)).
+  * If using RVM you have to do: `rvm use ruby-2.6.5`
+* Make sure `ruby --version` reports the correct version
+* Run `bundle install --path vendor/bundle` in the project root, `./android/` and `./ios/`
```
Comment (klinzo) [inline docs/08-cicd.md:37]:
```suggestion
* Run `bundle install --path vendor/bundle` in the project root AND in `./android/` AND in `./ios/`
```

File: docs/08-cicd.md [hunk 35]
```diff
@@ -1,80 +1,154 @@
...
 
 ### Metadata
 
-bundle exec fastlane deliver
+The CI/CD pipeline uploads and overwrites metadata during the delivery step.
+You can read more about managing metadata for iOS [here](https://docs.fastlane.tools/actions/deliver/).
+
+### Authenticating
+
+Authentication happens by setting the `FASTLANE_USER` and `FASTLANE_PASSWORD` environment variables as documented [above](#ios-variables). For more information visit the documentation [here](https://docs.fastlane.tools/best-practices/continuous-integration/#environment-variables-to-set) and read about [2-step-verification here](https://github.com/fastlane/fastlane/blob/ee904cd332ab38ca7c1979f0ab58f9389a51fb2d/spaceship/README.md#2-step-verification).
```
Comment (klinzo) [inline docs/08-cicd.md:141]: Is this only the case for ios?

File: docs/08-cicd.md [hunk 36]
```diff
@@ -1,80 +1,154 @@
...
 
-### Authenticating via spaceauth
-https://docs.fastlane.tools/best-practices/continuous-integration/#environment-variables-to-set
-https://github.com/fastlane/fastlane/tree/master/spaceship#2-step-verification
 
 ## BrowserStack
 
+We are using BrowserStack to run our E2E tests on real iOS and Android devices.
+The general documentation about [E2E tests and BrowserStack can be found here](04-e2e.md).
```
Comment (klinzo) [inline docs/08-cicd.md:147]:
```suggestion
The general documentation about E2E tests and BrowserStack can be found [here](04-e2e.md). 
```

File: docs/09-windows-setup.md [hunk 37]
```diff
@@ -0,0 +1,31 @@
...
+
+  # if you have a ultimate licence for intellij
+  choco install intellijidea-ultimate
+  # if you don't have the ultimate licence
+  choco install intellijidea-community
+````
+
+Reboot after the installation.
+
+## Windows Defender Settings
```
Comment (klinzo) [inline docs/09-windows-setup.md:29]:
```suggestion
## Troubleshooting
```

File: docs/09-windows-setup.md [hunk 38]
```diff
@@ -0,0 +1,31 @@
...
+
+Have a look to our [trouble shooting guide](02-troubleshooting.md) as well to fix some common windows issues.
```
Comment (klinzo) [inline docs/09-windows-setup.md:31]:
```suggestion
Have a look at our [trouble shooting guide](02-troubleshooting.md) as well for information on how to fix some common windows issues.
```

File: docs/10-apple-certifcates.md [hunk 39]
```diff
@@ -0,0 +1,30 @@
...
+* Via App Store
+* Via XCode
+
+Development certificates are for developers who want to install the app on their devices. You have to select the devices in the Provisioning Profile at [developer.apple.com](https://developer.apple.com/account/resources/profiles/list) you want to install the app on. This means the Device ID (UDID) has to be whitelisted.
+
+Ad-hoc certificates are for public tests with a small amount of users. The difference is that you can provide the testers a URL which they can open on their device to install it. The Device ID (UDID) still has to be whitelisted.
+
+Enterprise certificates allow you to deliver you app within your company or have a bigger public test. It is not possible for small organisation to acquire such a certificate.
+
+App Store certificates allows the upload of the app into the App Store. It is not possible to install a app store signed app directly on your device. It has to be uploaded to App Store Connect.
```
Comment (klinzo) [inline docs/10-apple-certifcates.md:30]: You could add `whitelisting required`, `installable by url`, `installable directly` to the table above. Not necessary though.

File: fastlane/Fastfile [hunk 40]
```diff
@@ -84,7 +84,7 @@ private_lane :generic_sentry_upload do |options|
   )
 end
 
-desc "Bump version file in Git repository"
+desc "Bump version and commits the new version in the GitHub repository"
```
Comment (klinzo) [inline fastlane/Fastfile:87]:
```suggestion
desc "Bump version and commit the new version in the GitHub repository"
```

File: fastlane/Fastfile [hunk 41]
```diff
@@ -113,7 +113,7 @@ platform :android do
     )
   end
 
-  desc "Run E2E tests on BrowserStack"
+  desc "Upload the APK and runs E2E tests on BrowserStack"
```
Comment (klinzo) [inline fastlane/Fastfile:116]:
```suggestion
  desc "Upload the APK and run E2E tests on BrowserStack"
```

File: fastlane/Fastfile [hunk 42]
```diff
@@ -207,7 +207,7 @@ platform :ios do
...
```
Comment (klinzo) [inline fastlane/Fastfile:210]:
```suggestion
  desc "Upload the APK and run E2E tests on BrowserStack"
```

File: fastlane/README.md [hunk 43]
```diff
@@ -19,7 +19,7 @@ or alternatively using `brew cask install fastlane`
 ```
 fastlane github_bump_version
 ```
-Bump version file in Git repository
+Bump version and commits the new version in the GitHub repository
```
Comment (klinzo) [inline fastlane/README.md:22]:
```suggestion
Bump version and commit the new version in the GitHub repository
```

File: fastlane/README.md [hunk 44]
```diff
@@ -28,27 +28,27 @@ Bump version file in Git repository
...
 fastlane android browserstack_upload_live
 ```
-Upload to Browserstack Live
+Upload the APK to BrowserStack Live
 ### android browserstack_e2e_tests
 ```
 fastlane android browserstack_e2e_tests
 ```
-Run E2E tests on BrowserStack
+Upload the APK and runs E2E tests on BrowserStack
```
Comment (klinzo) [inline fastlane/README.md:36]:
```suggestion
Upload the APK and run E2E tests on BrowserStack
```

File: fastlane/README.md [hunk 45]
```diff
@@ -57,12 +57,12 @@ Notify about the status of this build
...
 fastlane ios browserstack_upload_live
 ```
-Upload to Browserstack Live
+Upload the IPA to BrowserStack Live
 ### ios browserstack_e2e_tests
 ```
 fastlane ios browserstack_e2e_tests
 ```
-Run E2E tests on BrowserStack
+Upload the APK and runs E2E tests on BrowserStack
```
Comment (klinzo) [inline fastlane/README.md:65]:
```suggestion
Upload the APK and run E2E tests on BrowserStack
```

File: README.md [hunk 46]
```diff
@@ -1,108 +1,80 @@
...
 
-* Run `yarn start` to start the bundler
+### Prerequisites
+* Install [nodejs](https://nodejs.org/). At least v6 is required, but we recommend the v12 LTS.
+Using the latest version (v13) may lead to errors.
+* Install [yarn](https://yarnpkg.com/).
 
 #### Android
+* Install the Android SDK by using the [Android Support plugin](https://plugins.jetbrains.com/plugin/1792-android-support) in IntelliJ.
+* Install the latest stable SDK Platform and Android SDK Tools in the SDK Manager (Settings > Appearance & Behaviour > System Settings > Android SDK)
```
[the remark this answers is not in the document]
Comment (maxammann) [reply README.md:28]: Yeah dunno :D

File: docs/06-manual-builds.md [hunk 47]
```diff
@@ -0,0 +1,161 @@
...
+```
+
+Finally, install the certificates locally:
+
+```bash
+cd ios/
+bundle exec fastlane certificates
+```
+
+*Note: After setting up the certificates you can also start to use XCode to build the app.*
```
[the remark this answers is not in the document]
Comment (maxammann) [reply docs/06-manual-builds.md:38]: This is a good tip imo

File: docs/06-manual-builds.md (hunk 31 above)
Comment (maxammann) [reply docs/06-manual-builds.md:123]: These are valuable tricks to quickly restart the app

File: docs/08-cicd.md (hunk 35 above)
Comment (maxammann) [reply docs/08-cicd.md:141]: yes

File: docs/10-apple-certifcates.md (hunk 39 above)
Comment (maxammann) [reply docs/10-apple-certifcates.md:30]: That would bloat the table imo and would not replace the text below.

File: README.md [hunk 48]
```diff
@@ -1,108 +1,80 @@
...
+*Note: For setting the project up on Windows you may want to use Chocolatey and [follow this guide](docs/09-windows-setup.md).*
 
-* Run `yarn start` to start the bundler
+### Prerequisites
+* Install [nodejs](https://nodejs.org/). At least v6 is required, but we recommend the v12 LTS.
+Using the latest version (v13) may lead to errors.
+* Install [yarn](https://yarnpkg.com/).
 
 #### Android
+* Install the Android SDK by using the [Android Support plugin](https://plugins.jetbrains.com/plugin/1792-android-support) in IntelliJ.
```
[the remark this answers is not in the document]
Comment (maxammann) [reply README.md:27]: I added IntelliJ to above

File: docs/06-manual-builds.md [hunk 49]
```diff
@@ -0,0 +1,161 @@
...
+export FASTLANE_PASSWORD=<secret>
+export MATCH_PASSWORD=<secret>
+```
+
+Finally, install the certificates locally:
+
+```bash
+cd ios/
+bundle exec fastlane certificates
+```
```
[the remark this answers is not in the document]
Comment (maxammann) [reply docs/06-manual-builds.md:38]: I favor not to use bullet lists here as I never did so far

File: docs/06-manual-builds.md (hunk 30 above)
Comment (maxammann) [reply docs/06-manual-builds.md:106]: I'd keep these separate as it makes clear that the first step decrypts the keystore and is not required for the `fastlane keystore` command

File: docs/06-manual-builds.md (hunk 32 above)
Comment (maxammann) [reply docs/06-manual-builds.md:111]: I restructured the section

File: docs/06-manual-builds.md (hunk 30 above)
Comment (maxammann) [reply docs/06-manual-builds.md:106]: I restructured the section

File: docs/06-manual-builds.md [hunk 50]
```diff
@@ -0,0 +1,161 @@
...
+bundle exec pod install
+bundle exec fastlane build
+```
+
+Fastlane should report where the build artifacts are. These can be uploaded to App Store Connect or distributed via another way.
+
+## Manually building for Android
+
+If you do not have Fastlane installed [you can skip to the next section](#quick-builds-using-a-test-signing-key), which does not require FL.
+
```
[the remark this answers is not in the document]
Comment (maxammann) [reply docs/06-manual-builds.md:80]: I restructured the section
